crates/core/component/funding/src/component/liquidity_tournament/nullifier/mod.rs

```diff
@@ -0,0 +1,47 @@
+use async_trait::async_trait;
+use penumbra_sdk_txhash::TransactionId;
+
+use crate::component::state_key;
+use cnidarium::{StateRead, StateWrite};
+use penumbra_sdk_proto::{StateReadProto, StateWriteProto};
+use penumbra_sdk_sct::{component::clock::EpochRead, Nullifier};
+
+#[allow(dead_code)]
+#[async_trait]
+pub trait NullifierRead: StateRead {
+    /// Gets the transaction id associated with the given nullifier from the JMT.
+    async fn get_txid_from_nullifier(&self, nullifier: Nullifier) -> Option<TransactionId> {
```

Comment:
Would rename to `lqt_spent_nullifier` because the important load bearing way this method is used is to check membership into a set, and avoid duplicate votes. I would add a comment in the rust doc that explains that if the nullifier has been spent the `TransactionId` is returned, otherwise `None` is returned.